libraries/libfc/include/fc/crypto/bls_public_key.hpp

```diff
@@ -3,43 +3,67 @@
 #include <fc/reflect/reflect.hpp>
 #include <fc/reflect/variant.hpp>
 #include <bls12-381/bls12-381.hpp>
+#include <algorithm>
 
 namespace fc::crypto::blslib {
 
    namespace config {
       const std::string bls_public_key_prefix = "PUB_BLS_";
    };
 
-   class bls_public_key
-   {
-      public:
- 
-         bls_public_key() = default;
-         bls_public_key( bls_public_key&& ) = default;
-         bls_public_key( const bls_public_key& ) = default;
-         explicit bls_public_key( const bls12_381::g1& pkey ) {_pkey = pkey;}
-         // affine non-montgomery base64url with bls_public_key_prefix
-         explicit bls_public_key(const std::string& base64urlstr);
+   // Immutable after construction.
+   // Provides an efficient wrapper around bls12_381::g1.
+   // Serialization form:
+   //   Non-Montgomery form and little-endian encoding for the field elements.
+   //   Affine form for the group element (the z component is 1 and not included in the serialization).
+   //   Binary serialization encodes x component first followed by y component.

```

Comment:
Please update the comment with the explanation from issue #2334 describing why we use this internal representation and this form for serialization.